tests/rptest/services/redpanda.py

```diff
@@ -832,6 +833,188 @@ def __init__(self,
+        nodes where serialized startup becomes annoying, pass parallel=True.
+
+        :param parallel: if true, run clean and start operations in parallel
+                         for the nodes being started.
+        :param expect_fail: if true, expect redpanda nodes to terminate shortly
+                            after starting.  Raise exception if they don't.
+        """
+        pass
+
+    def wait_until(self, fn, timeout_sec, backoff_sec, err_msg=None):
+        """
+        Cluster-aware variant of wait_until, which will fail out
+        early if a node dies.
+
+        This is useful for long waits, which would otherwise not notice
+        a test failure until the end of the timeout, even if redpanda
+        already crashed.
+        """
+        pass
+
+    def get_node_memory_mb(self):
+        pass
+
+    def get_node_cpu_count(self):
+        pass
+
+    def get_node_disk_free(self):
+        pass
+
+    def set_resource_settings(self, rs):
+        pass
+
+    def lsof_node(self, node: ClusterNode, filter: Optional[str] = None):
+        """
+        Get the list of open files for a running node
+
+        :param filter: If given, this is a grep regex that will filter the files we list
+
+        :return: yields strings
+        """
+        pass
+
+    def set_si_settings(self, si_settings: SISettings):
+        pass
+
+    def add_extra_rp_conf(self, conf):
+        pass
+
+    def storage(self, all_nodes: bool = False, sizes: bool = False):
+        """
+        :param all_nodes: if true, report on all nodes, otherwise only report
+                          on started nodes.
+
+        :returns: instances of ClusterStorage
+        """
+        pass
+
+    def set_extra_rp_conf(self, conf):
+        pass
+
+    def metrics_sample(
+        self,
+        sample_pattern,
+        nodes=None,
+        metrics_endpoint: MetricsEndpoint = MetricsEndpoint.METRICS,
+    ) -> Optional[MetricSamples]:
+        """
+        Query metrics for a single sample using fuzzy name matching. This
+        interface matches the sample pattern against sample names, and requires
+        that exactly one (family, sample) match the query. All values for the
+        sample across the requested set of nodes are returned in a flat array.
+
+        None will be returned if less than one (family, sample) matches.
+        An exception will be raised if more than one (family, sample) matches.
+
+        Example:
+
+           The query:
+
+              redpanda.metrics_sample("under_replicated")
+
+           will return an array containing MetricSample instances for each node and
+           core/shard in the cluster. Each entry will correspond to a value from:
+
+              family = vectorized_cluster_partition_under_replicated_replicas
+              sample = vectorized_cluster_partition_under_replicated_replicas
+        """
+        pass
+
+    def signal_redpanda(self, node, signal=signal.SIGKILL, idempotent=False):
```

Comment:
I'm not sure signal is really the portable primitive here: we usually want to stop redpanda (either hard or soft), or to do something funky like SIGSTOP'ing (pausing) the process.

For the stopping cases, this should be a stop_node thing so that it can do whatever kubernetes business is needed (like tainting a node so that it doesn't immediately restart), and for the SIGSTOP case that maybe needs to be some explicit method that can be clearly not-implemented on some backends.

Reply:
i've removed `signal_redpanda` from the base class with 20698a24773acd79825a312d6a642af447ace67e because it looks like `stop_node` has a `forced` arg that is implemented in `RedpandaService` as `SIGKILL`